### DISCUSSION TITLE
RaspberryPi ZeroW/DietPi/Room-Assistant solution for 2.15.1 out of memory errors

### DISCUSSION THREAD

**Discussion by e-hollebone** [mKeRix/room-assistant#614] [Show and tell]
For those running out of memory with installing or updating to 2.15.1 on RaspberryPi ZeroWs or that want to try an alternative to the current RaspberryPi minimal image, I got Room Assistant running on [dietPi](https://dietpi.com/) (https://dietpi.com/) on the RPI ZeroW. 

[tl:dr]
I was having trouble upgrading the RPI ZeroW from 2.15.0 to 2.15.1 and was running into memory allocation errors just as npm was running the compile portion of the install.  Nothing I could find helped. See below for the exact error:

```
npm timing reify:retireShallow Completed in 125ms
[         .........] / idealTree: timing reify:retireShallow Completed in 125ms
<--- Last few GCs --->

[9300:0x542ca38]   408281 ms: Mark-sweep 125.2 (129.2) -> 124.9 (129.2) MB, 9827.1 / 29.8 ms  (average mu = 0.285, current mu = 0.059) allocation failure scavenge might not succeed
[9300:0x542ca38]   417170 ms: Mark-sweep 125.0 (129.2) -> 125.0 (129.4) MB, 8865.1 / 15.2 ms  (average mu = 0.161, current mu = 0.003) allocation failure GC in old space requested

<--- JS stacktrace --->

FATAL ERROR: MarkCompactCollector: young object promotion failed Allocation failed - JavaScript heap out of memory
Aborted

```
Check the available memory, with the out-of-the-box raspberry minimal image, the **stable memory consumption is 430MB** available. I have five RPI ZeroWs, and they all report the same. I’ll say that is a great thing from a consistency perspective.

Don’t be led down the rabbit hole with increasing swap, playing with vm.swappiness or try to allocate more heap space via  "--max_old_space_size=xxxx", none of those attempts worked

Don’t upgrade npm, npm@6.14.11 works just fine, npm@7.6.3 will work on rpi3 or 4
Don’t bother playing with the nodejs memory
[/td:dr]

What did work was moving away from the official minimal image and switching to dietPi. Using that configuration, the **available memory increased by 48MB to 478MB**. Awesome and it even boots faster, in about 40 seconds.

There is a very comprehensive guide to installing dietPi (https://dietpi.com/docs/install/) and some great videos. I chose to use the headless configuration (see [installation video #3](https://www.youtube.com/watch?v=vlMpn9u0Y4o) - https://www.youtube.com/watch?v=vlMpn9u0Y4o) and edited the dietpi.txt and dietpi-wifi.txt files to make the process automated, headless and using a static ip.

I made the following changes to the dietPi.txt file in the /boot directory. Please make sure to adapt these settings to your configuration as you see fit.

> ```
> AUTO_SETUP_ACCEPT_LICENSE=1
> AUTO_SETUP_LOCALE=en_CA.UTF-8
> AUTO_SETUP_KEYBOARD_LAYOUT=us
> AUTO_SETUP_TIMEZONE=America/Toronto
> AUTO_SETUP_NET_ETHERNET_ENABLED=0
> AUTO_SETUP_NET_WIFI_ENABLED=1
> AUTO_SETUP_NET_WIFI_COUNTRY_CODE=CA
> AUTO_SETUP_NET_USESTATIC=1
> > _Set to your specfic network parameters_
> AUTO_SETUP_NET_HOSTNAME=_<your hostname>_
> AUTO_SETUP_SSH_SERVER_INDEX=-2  <-- **ONLY if you want to use OpenSSH over dropbear**
> AUTO_SETUP_AUTOSTART_TARGET_INDEX=7
> AUTO_SETUP_AUTOSTART_LOGIN_USER=root
> AUTO_SETUP_AUTOMATED=1
> AUTO_SETUP_GLOBAL_PASSWORD=_<your password>_
> AUTO_SETUP_INSTALL_SOFTWARE_ID=105 #OpenSSHServer
> AUTO_SETUP_INSTALL_SOFTWARE_ID=152 #Avahi-daemon
> CONFIG_BOOT_WAIT_FOR_NETWORK=2
> CONFIG_NTP_MODE=2
> CONFIG_ENABLE_IPV6=0
> ```

Update dietpi-wifi.txt with your wifi credentials. Save both files. 

Eject the SD card and insert it into your RPI ZeroW and boot.

It will take some time before the system is available, for me it took on the order of 15 minutes, with the APT update being the most time-consuming. Once the initial install is complete, add the Bluetooth libs and tools via the dietpi-config tool (similar to raspi-config)

> `sudo dietpi-config -> advanced -> bluetooth: change to on`

Note: also turn on i2c, uart, gpio etc. as appropriate if you're going to be adding sensors to room-assistant. I have only tested the BluetoothLE and home-assistant integrations of this implementation, I don't have any means to verify if any of the other sensors or integrations require additional packages so users be aware that you may have to do additional detective work to make those parts functional.

Finally, use apt to add the minimum required Debian packages.

Make sure apt is up to date 

> `sudo apt update`

Install python, avahi-utils, libnss-mdns and RA build requirements 

> `sudo apt -y install python3-minimal build-essential libavahi-compat-libdnssd-dev bluetooth libbluetooth-dev libudev-dev avahi-utils libnss-mdns`

Optionally, I chose to add lsb-release and some other libs for some personal scripting I do.

>` sudo apt -y install lsb-release dnsutils apt-utils`

Add dietpi user to systemd-journal group if you’re going to run room-assistant as a daemon

>` sudo addgroup dietpi systemd-journal`

At this point, you're done the heavy lifting, so for good measure, I do a reboot and then setup my ssh keys and user environment.

Now you can install room-assistant as per the normal instructions from the [Installing room-assistant on an RPI ZeroW](https://www.room-assistant.io/guide/quickstart-pi-zero-w.html#installing-room-assistant) (https://www.room-assistant.io/guide/quickstart-pi-zero-w.html#installing-room-assistant) in the RaspberryPi Zero installation guide but you can skip the APT step as it was already included above section but it won't hurt anything if you do include it.

PS: The default user will be **dietpi** and not just pi. This will affect the room-assistant.service daemon file example in /etc/systemd/system used for the autostart. You'll have to either update the user and the working directory entries in the example or create the pi user (don't forget to add the pi user to the systemd-journal group if you go that route).

PPS: More notes and cautions. I have not tested this with Ansible yet either. I will be doing that in the future and I will update this post if there are any additional requirements to get Ansible functional.

**Comment by AML225** [comment]
I really appreciate you posting this as I was running into this exact issue and getting frustrated. I followed your instructions however I am getting the following errors that I haven't been able to resolve:

```
pi@livingroom:~/room-assistant $ room-assistant
[Nest] 673   - 03/21/2021, 9:44:56 PM   [ClusterService] Cannot find module 'mdns'
Require stack:
- /opt/nodejs/lib/node_modules/room-assistant/dist/cluster/cluster.service.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/entities/entities.service.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/entities/entities.module.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/app.module.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/main.js
- /opt/nodejs/lib/node_modules/room-assistant/bin/room-assistant.js
Error: Cannot find module 'mdns'
Require stack:
- /opt/nodejs/lib/node_modules/room-assistant/dist/cluster/cluster.service.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/entities/entities.service.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/entities/entities.module.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/app.module.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/main.js
- /opt/nodejs/lib/node_modules/room-assistant/bin/room-assistant.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/opt/nodejs/lib/node_modules/room-assistant/dist/cluster/cluster.service.js:45:12)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
3/21/2021, 9:45:04 PM - info - IntegrationsModule: Loading integrations: home-assistant, bluetooth-classic, bluetooth-low-energy
3/21/2021, 9:45:08 PM - info - NestFactory: Starting Nest application...
3/21/2021, 9:45:09 PM - error - ExceptionHandler: Cannot find module '@mkerix/noble'
Require stack:
- /opt/nodejs/lib/node_modules/room-assistant/dist/integration-support/bluetooth/bluetooth.service.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/integrations/bluetooth-classic/bluetooth-classic.service.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/integrations/bluetooth-classic/bluetooth-classic.module.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/integrations/integrations.module.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/app.module.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/main.js
- /opt/nodejs/lib/node_modules/room-assistant/bin/room-assistant.js
```

Here is my config:
```
cluster:
  autoDiscovery: false
  networkInterface: wlan0
  port: 6425
  peerAddresses:
    - 192.168.1.24:6425
    - 192.168.1.31:6425
  weight: 10
  quorum: 2
global:
  instanceName: livingroom
  integrations:
    - homeAssistant
    - bluetoothClassic
    - bluetoothLowEnergy
homeAssistant:
  mqttUrl: 'mqtt://192.168.1.9:1883'
  mqttOptions:
    username: mqtt_user
    password: *censored*
bluetoothClassic:
  addresses:
    - 'A0:FB:C5:DB:19:65'
    - 'B4:56:E3:CA:E1:81'
bluetoothLowEnergy:
  whitelist:
    - 7777772E6B6B6D636E2E636F6D000001-0-0
  onlyIBeacon: true
  timeout: 60
entities:
  behaviors:
    ble-7777772e6b6b6d636e2e636f6d000001-0-0-tracker:
      debounce:
        wait: 5
        maxWait: 10
      rollingAverage:
        window: 10

**Reply by e-hollebone** [reply]
Your error has nothing to do with your room-assistant yaml files (by the way, your config looks fine). You're missing one or more underlying OS modules that Room-Assistant depends on.  In your case, looks like avahi (bonjour) is missing. 

Did you add the add extra software while editing the dietpi.txt stage?
> AUTO_SETUP_INSTALL_SOFTWARE_ID=152 #Avahi-daemon

No worries, you can correct this using the dietip-software utility and look for avahi-daemon. If I remember, it's in the menu under Software Additional -> #152 Avahi-Daemon. 

ps run dietpi-software as root or use sudo

**Reply by AML225** [reply]
> Your error has nothing to do with your room-assistant yaml files (by the way, your config looks fine). You're missing one or more underlying OS modules that Room-Assistant depends on.  In your case, looks like avahi (bonjour) is missing. 
> 
> 
> 
> Did you add the add extra software while editing the dietpi.txt stage?
> 
> > AUTO_SETUP_INSTALL_SOFTWARE_ID=152 #Avahi-daemon
> 
> 
> 
> No worries, you can correct this using the dietip-software utility and look for avahi-daemon. If I remember, it's in the menu under Software Additional -> #152 Avahi-Daemon. 
> 
> 
> 
> ps run dietpi-software as root or use sudo

I did include the #152 Avahi-Daemon in the dietpi.txt and just confirmed that it is installed with dietpi-software. Not sure what's going on.

**Reply by e-hollebone** [reply]
the other problem for me was libnss-mdns was missing, can you check that with
> ```sudo apt list --installed | grep mdns```

**Reply by AML225** [reply]
Confirmed that libnss-mdns is installed and the latest version. I even tried starting from scratch and re-installing dietpi. Still getting the same errors. Cannot find module 'mdns' and cannot find module '@mkerix/noble' when I try to start room-assistant.

**Reply by judy** [reply]
I have these issues too. I ran the recommended check and got `libnss-mdns/stable,now 0.14.1-1+b5 armhf [installed]`. Full terminal log:

```shell
dietpi@StudioRoomAssistant:~/room-assistant $ room-assistant
[Nest] 4141   - 04/14/2021, 4:58:54 PM   [ClusterService] Cannot find module 'mdns'
Require stack:
- /opt/nodejs/lib/node_modules/room-assistant/dist/cluster/cluster.service.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/entities/entities.service.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/entities/entities.module.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/app.module.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/main.js
- /opt/nodejs/lib/node_modules/room-assistant/bin/room-assistant.js
Error: Cannot find module 'mdns'
Require stack:
- /opt/nodejs/lib/node_modules/room-assistant/dist/cluster/cluster.service.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/entities/entities.service.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/entities/entities.module.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/app.module.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/main.js
- /opt/nodejs/lib/node_modules/room-assistant/bin/room-assistant.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/opt/nodejs/lib/node_modules/room-assistant/dist/cluster/cluster.service.js:45:12)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
4/14/2021, 4:58:59 PM - info - IntegrationsModule: Loading integrations: home-assistant, bluetooth-classic
4/14/2021, 4:59:02 PM - info - NestFactory: Starting Nest application...
4/14/2021, 4:59:03 PM - error - ExceptionHandler: Cannot find module '@mkerix/noble'
Require stack:
- /opt/nodejs/lib/node_modules/room-assistant/dist/integration-support/bluetooth/bluetooth.service.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/integrations/bluetooth-classic/bluetooth-classic.service.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/integrations/bluetooth-classic/bluetooth-classic.module.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/integrations/integrations.module.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/app.module.js
- /opt/nodejs/lib/node_modules/room-assistant/dist/main.js
- /opt/nodejs/lib/node_modules/room-assistant/bin/room-assistant.js
```

**Reply by mKeRix** [reply]
Note that once you install the system dependency you need to re-run the installation for room-assistant, otherwise the room-assistant dependency still won't be there.

**Reply by judy** [reply]
I got it working! 

> Optionally, I chose to add lsb-release and some other libs for some personal scripting I do.

```
sudo apt -y install lsb-release dnsutils apt-utils
```

I didn't run that the first time. I did this time, then re-ran `sudo npm i --global --unsafe-perm room-assistant`. It appears to be working now! One of those "optional" libs isn't so optional.

**Comment by mKeRix** [comment]
Thanks for your detailed write-up on this topic! I'll pin the discussion for more visibility. For me staying at npm v6 (`npm install -g npm@6`) works fine on the Pi Zero Ws, but as npm v7 is now triggering upgrade notifications for everyone I'd like to find a way around these constraints for regular Raspbian users as well. I'll try to investigate this a bit more sometime soon-ish.

**Reply by mKeRix** [reply]
I looked at this for a little bit today... unfortunately with no satisfactory outcome. By trimming out a few dependencies I could reduce the tree enough that an install without any memory errors was possible using npm v7, however a clean install ended up taking 36 minutes - which is just too long imo.

Overview of the options I looked at:
- npm v6 - works fine as before, but will at some point become legacy
- npm v7 - installation finishes after trimming dependencies, but takes a long time (36min)
- yarn - fails during installation
- pnpm - does not seem to be meant for this use case, no way to use the binary (installs fine though)
- Docker - works fine

At the moment I'm considering whether it would make more sense to switch Pi Zeros over to the Docker option, as the installation process there is much faster and more consistent. Since the npm installation process was just re-written in v7 I don't think they'll optimize for our use cases, the focus seems to be on other scenarios. Users that followed the getting started guide or that use Ansible should be able to switch seamlessly, as the configs can just be mounted into the container.

**Reply by mKeRix** [reply]
Since v2.15.1 shouldn't have this issue anymore I'll unpin this again. I'm still unsure what the future of the installation experience will be, but this takes the pressure off a bit.

**Comment by carsaig** [comment]
> For those running out of memory with installing or updating to 2.15.1 on RaspberryPi ZeroWs or that want to try an alternative to the current RaspberryPi minimal image, I got Room Assistant running on [dietPi](https://dietpi.com/) (https://dietpi.com/) on the RPI ZeroW.
> 
> [tl:dr] I was having trouble upgrading the RPI ZeroW from 2.15.0 to 2.15.1 and was running into memory allocation errors just as npm was running the compile portion of the install. Nothing I could find helped. See below for the exact error:
> 
> ```
> npm timing reify:retireShallow Completed in 125ms
> [         .........] / idealTree: timing reify:retireShallow Completed in 125ms
> <--- Last few GCs --->
> 
> [9300:0x542ca38]   408281 ms: Mark-sweep 125.2 (129.2) -> 124.9 (129.2) MB, 9827.1 / 29.8 ms  (average mu = 0.285, current mu = 0.059) allocation failure scavenge might not succeed
> [9300:0x542ca38]   417170 ms: Mark-sweep 125.0 (129.2) -> 125.0 (129.4) MB, 8865.1 / 15.2 ms  (average mu = 0.161, current mu = 0.003) allocation failure GC in old space requested
> 
> <--- JS stacktrace --->
> 
> FATAL ERROR: MarkCompactCollector: young object promotion failed Allocation failed - JavaScript heap out of memory
> Aborted
> ```
> 
> Check the available memory, with the out-of-the-box raspberry minimal image, the **stable memory consumption is 430MB** available. I have five RPI ZeroWs, and they all report the same. I’ll say that is a great thing from a consistency perspective.
> 
> Don’t be led down the rabbit hole with increasing swap, playing with vm.swappiness or try to allocate more heap space via "--max_old_space_size=xxxx", none of those attempts worked
> 
> Don’t upgrade npm, npm@6.14.11 works just fine, npm@7.6.3 will work on rpi3 or 4 Don’t bother playing with the nodejs memory [/td:dr]
> 
> What did work was moving away from the official minimal image and switching to dietPi. Using that configuration, the **available memory increased by 48MB to 478MB**. Awesome and it even boots faster, in about 40 seconds.
> 
> There is a very comprehensive guide to installing dietPi (https://dietpi.com/docs/install/) and some great videos. I chose to use the headless configuration (see [installation video #3](https://www.youtube.com/watch?v=vlMpn9u0Y4o) - https://www.youtube.com/watch?v=vlMpn9u0Y4o) and edited the dietpi.txt and dietpi-wifi.txt files to make the process automated, headless and using a static ip.
> 
> I made the following changes to the dietPi.txt file in the /boot directory. Please make sure to adapt these settings to your configuration as you see fit.
> 
> > ```
> > AUTO_SETUP_ACCEPT_LICENSE=1
> > AUTO_SETUP_LOCALE=en_CA.UTF-8
> > AUTO_SETUP_KEYBOARD_LAYOUT=us
> > AUTO_SETUP_TIMEZONE=America/Toronto
> > AUTO_SETUP_NET_ETHERNET_ENABLED=0
> > AUTO_SETUP_NET_WIFI_ENABLED=1
> > AUTO_SETUP_NET_WIFI_COUNTRY_CODE=CA
> > AUTO_SETUP_NET_USESTATIC=1
> > > _Set to your specfic network parameters_
> > AUTO_SETUP_NET_HOSTNAME=_<your hostname>_
> > AUTO_SETUP_SSH_SERVER_INDEX=-2  <-- **ONLY if you want to use OpenSSH over dropbear**
> > AUTO_SETUP_AUTOSTART_TARGET_INDEX=7
> > AUTO_SETUP_AUTOSTART_LOGIN_USER=root
> > AUTO_SETUP_AUTOMATED=1
> > AUTO_SETUP_GLOBAL_PASSWORD=_<your password>_
> > AUTO_SETUP_INSTALL_SOFTWARE_ID=105 #OpenSSHServer
> > AUTO_SETUP_INSTALL_SOFTWARE_ID=152 #Avahi-daemon
> > CONFIG_BOOT_WAIT_FOR_NETWORK=2
> > CONFIG_NTP_MODE=2
> > CONFIG_ENABLE_IPV6=0
> > ```
> 
> Update dietpi-wifi.txt with your wifi credentials. Save both files.
> 
> Eject the SD card and insert it into your RPI ZeroW and boot.
> 
> It will take some time before the system is available, for me it took on the order of 15 minutes, with the APT update being the most time-consuming. Once the initial install is complete, add the Bluetooth libs and tools via the dietpi-config tool (similar to raspi-config)
> 
> > `sudo dietpi-config -> advanced -> bluetooth: change to on`
> 
> Note: also turn on i2c, uart, gpio etc. as appropriate if you're going to be adding sensors to room-assistant. I have only tested the BluetoothLE and home-assistant integrations of this implementation, I don't have any means to verify if any of the other sensors or integrations require additional packages so users be aware that you may have to do additional detective work to make those parts functional.
> 
> Finally, use apt to add the minimum required Debian packages.
> 
> Make sure apt is up to date
> 
> > `sudo apt update`
> 
> Install python, avahi-utils, libnss-mdns and RA build requirements
> 
> > `sudo apt -y install python3-minimal build-essential libavahi-compat-libdnssd-dev bluetooth libbluetooth-dev libudev-dev avahi-utils libnss-mdns`
> 
> Optionally, I chose to add lsb-release and some other libs for some personal scripting I do.
> 
> > ` sudo apt -y install lsb-release dnsutils apt-utils`
> 
> Add dietpi user to systemd-journal group if you’re going to run room-assistant as a daemon
> 
> > ` sudo addgroup dietpi systemd-journal`
> 
> At this point, you're done the heavy lifting, so for good measure, I do a reboot and then setup my ssh keys and user environment.
> 
> Now you can install room-assistant as per the normal instructions from the [Installing room-assistant on an RPI ZeroW](https://www.room-assistant.io/guide/quickstart-pi-zero-w.html#installing-room-assistant) (https://www.room-assistant.io/guide/quickstart-pi-zero-w.html#installing-room-assistant) in the RaspberryPi Zero installation guide but you can skip the APT step as it was already included above section but it won't hurt anything if you do include it.
> 
> PS: The default user will be **dietpi** and not just pi. This will affect the room-assistant.service daemon file example in /etc/systemd/system used for the autostart. You'll have to either update the user and the working directory entries in the example or create the pi user (don't forget to add the pi user to the systemd-journal group if you go that route).
> 
> PPS: More notes and cautions. I have not tested this with Ansible yet either. I will be doing that in the future and I will update this post if there are any additional requirements to get Ansible functional.

Thanks for the details. I got it up and running, though not the service. Which ExecStart path variable did you use to point to room-assistant?
This is my service file:

`[Unit]
Description=room-assistant service

[Service]
Type=notify
ExecStart=/usr/local/lib/node_modules/room-assistant/dist/integrations/home-assistant
WorkingDirectory=/root/room-assistant
TimeoutStartSec=120
TimeoutStopSec=30
Restart=always
RestartSec=10
WatchdogSec=60
User=dietpi

[Install]
WantedBy=multi-user.target`

The service fails with the error:
`systemd[1176]: room-assistant.service: Changing to the requested working directory failed: Permission denied
 systemd[1176]: room-assistant.service: Failed at step CHDIR spawning /usr/local/lib/node_modules/room-assistant: Permission denied`

Any suggestions?
